TrafficCapture/dockerSolution/build.gradle

```diff
@@ -25,89 +21,108 @@ dependencies {
-                    projectName, "src/main/docker/${escapedProjectName}")
-            dependencies +=stageConsoleDependenciesFromApplications(project as Project,
-                    List.of(project(":CreateSnapshot"),
-                            project(":MetadataMigration"),
-                    ), projectName, "src/main/docker/${escapedProjectName}")
-
+            // Define lists of libraries and applications
+            def libraries = [
+                project(":libraries:kafkaCommandLineFormatter")
+            ]
 
-            def parentCopyTask = project.tasks.create("copyArtifact_${projectName}")
-            dependencies.forEach { d -> parentCopyTask.dependsOn(d) }
+            def applications = [
+                project(":CreateSnapshot"),
+                project(":MetadataMigration")
+            ]
 
-            dependsOn "copyArtifact_${escapedProjectName}"
+            // Create a single sync task to copy the required files
+            def destDir = "build/docker/${dockerImageName}_${escapedProjectName}"
+            def syncTask = project.tasks.create("syncArtifact_${dockerImageName}_${escapedProjectName}", Sync) {
+                into destDir
+                duplicatesStrategy = DuplicatesStrategy.EXCLUDE
+
+                // Copy libraries
+                libraries.each { libProject ->
+                    def applicationDestDir = "staging/${libProject.name}/"
+                    from (libProject.configurations.findByName("runtimeClasspath").files) {
+                        into "${applicationDestDir}/lib"
+                    }
+                    from (libProject.tasks.getByName('jar')) {
+                        into "${applicationDestDir}/lib"
+                    }
+                }
+
+                // Copy applications
+                applications.each { appProject ->
+                    def applicationDestDir = "staging/${appProject.name}/"
+                    from (appProject.configurations.findByName("runtimeClasspath").files) {
+                        into "${applicationDestDir}/lib"
+                    }
+                    from (appProject.tasks.getByName('jar')) {
+                        into "${applicationDestDir}/lib"
+                    }
+                    from (appProject.tasks.getByName('startScripts').outputs.files) {
+                        into "${applicationDestDir}/bin"
+                    }
+                }
+
+                from "src/main/docker/${projectName}"
+            }
+
+            // Collect assemble tasks from libraries and applications
+            def assembleTasks = (libraries + applications).collect { it.tasks.named("assemble") }
+            // Ensure the sync task depends on the assemble tasks
+            syncTask.dependsOn(assembleTasks)
+
+            dependsOn syncTask
+            dependsOn assembleTasks
+            // Migration Console base image is the test console
             dependsOn "buildDockerImage_elasticsearchTestConsole"
```

Comment:
Would help my cognitive load if all this logic that pertains to the Migration Console was in a helper function